src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -169,7 +169,16 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
             else if (typeof(DSFunctionBase).IsAssignableFrom(type))
             {
                 var mangledName = obj["FunctionSignature"].Value<string>();
-                var functionDescriptor = libraryServices.GetFunctionDescriptor(mangledName);
+                var priorNames = libraryServices.GetPriorNames();
+                FunctionDescriptor functionDescriptor;
+
+                // Attempt to located a newer migrated version of the node
+                if (priorNames.ContainsKey(mangledName))
```

Comment:
Just use `TryGetValue `. You can check for the key and get the value at the key at the same time.

Reply:
I actually was using that originally but was hoping to figure out how to eliminate having to define `functionDescriptor` without assigning a value.  I ended up having it anyway so I agree and will switch back.  I also have a few other changes coming, this doesn't actually solve all the migration scenarios properly yet.

```C#
priorNames.TryGetValues(mangledName, out functionDescriptor)
```